src/core.ts

```diff
@@ -557,11 +583,17 @@ function TrackerCore(schemas: Schemas, sanitizeEvent: SanitizeEventFn): TrackerF
                     throw new Error('call autoSetVisitor() at least once before an event is sent');
                 }
 
-                (event as FullSendableEvent).clientVersion = { raw: internals.VERSION };
-                (event as FullSendableEvent).eventType = type;
-                (event as FullSendableEvent).customer = internals.CUSTOMER;
-                (event as FullSendableEvent).visit = internals.VISIT as SendableVisit;
-                return event as FullSendableEvent;
+                const metadata = internals.getPreparedMetadata(event.metadata);
+                const fullSendableEvent: FullSendableEvent = {
+                    ...event,
+                    clientVersion: { raw: internals.VERSION },
+                    eventType: type,
+                    customer: internals.CUSTOMER,
+                    visit: internals.VISIT as SendableVisit,
+                    ...metadata && { metadata }
```

Comment:
The metadata field must have an array value or not exist in the object. If it is undefined, `getRemovedFields` will treat this field as invalid and add { key: 'gbi-incorrect-field': 'metadata' } to the metadata.